core/src/main/java/hudson/model/UpdateSite.java

```diff
@@ -1079,6 +1088,19 @@ public boolean isForNewerHudson() {
             }
         }
 
+        /**
+         * @since TODO
+         * @return
+         */
+        public boolean isForNewerJava() {
+            try {
+                return requiredJava != null && new VersionNumber(requiredJava).isNewerThan(
+                        new VersionNumber(System.getProperty("java.version")));
```

Comment:
Needs a comment like above. I would probably move it to a static method in order to integrate with #2949

Reply:
as above, this is the wrong property